markovflow/kalman_filter.py

```diff
@@ -495,3 +494,119 @@ def _log_det_observation_precision(self):
     def observations(self):
         """ Observation vector """
         return self.sites.means
+
+
+@tf_scope_class_decorator
+class KalmanFilterWithSparseSites(BaseKalmanFilter):
+    r"""
+    Performs a Kalman filter on a :class:`~markovflow.state_space_model.StateSpaceModel`
+    and :class:`~markovflow.emission_model.EmissionModel`, with Gaussian sites, over a time grid.
+    """
+
+    def __init__(self, state_space_model: StateSpaceModel, emission_model: EmissionModel, sites: GaussianSites,
+                 time_grid_shape: tf.TensorShape, observations_index: tf.Tensor, observations: tf.Tensor):
+        """
+        :param state_space_model: Parameterises the latent chain.
+        :param emission_model: Maps the latent chain to the observations.
+        :param sites: Gaussian sites over the time grid.
+        :param time_grid_shape: Time grid shape, tuple (T, 1).
```

Comment:
If it is just (T,1) , maybe simpler to same the num_grid_points as an integer.

you can build the shape in the code -> shape = (num_grid_points, 1)
That makes the interface simpler.